gilda/process.py

```diff
@@ -203,3 +203,37 @@ def depluralize(word):
         return word[:-1], 'plural_s'
     # Note: there don't seem to be any compelling examples of -f or -fe -> ves
     # so it is not implemented
+
+
+def replace_roman_arabic(s):
+    for pattern, result in roman_arabic_patterns:
```

Comment:
if you precompile the patterns it will give a speed-up

Reply:
oh I see what's happening. They already are. You could change it to `pattern.sub(result, s)` I think. Not sure if that is a big deal or not

Reply:
I think the `re.sub` functions just check if it's a string, compile it, then delegate to the class

<img width="623" alt="Screen Shot 2021-10-07 at 17 41 59" src="https://user-images.githubusercontent.com/5069736/136418249-d42b99f7-b43d-4928-bbe3-a52eb4e0d904.png">

Reply:
So I guess you could save a few operations for calling a function, doing some conditionals and making some new frames for try/excepts. Not sure how big of an impact that would have, but I guess it couldn't hurt to optimize